addusers.py

```diff
@@ -470,7 +452,9 @@ def parse_rows(rows, select_user=None):
                     response = openstack.grant_role(auth_url, ks_project.id,
                                                     ks_user.id,
                                                     ks_member_role.id)
-                
+                    rows[content[project].row].extend([
```

Comment:
I don't think you should need to repeat this code block twice?  Can you put it outside the if statement?